.github/workflows/js_cdn_deploy.yaml

```diff
@@ -0,0 +1,176 @@
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        description: The name of the workspace to create the artifact for.
+        type: string
+        required: true
+      environment:
+        description: Environment where the artifact will be deployed.
+        type: string
+        required: true
+      resource_group_name:
+        description: Web App resource group name.
+        type: string
+        required: true
+      storage_account_name:
+        description: Storage account name.
+        type: string
+        required: true
+      profile_cdn_name:
+        description: Profile CDN name.
+        type: string
+        required: true
+      endpoint_name:
+        description: CDN endpoint name.
+        type: string
+        required: true
+      blob_container_name:
+        description: Blob container name.
+        type: string
+        required: true
+      custom_env_config_variables:
+        description: Contain variable for env-config.js file generation (Optional)
+        type: string
+        required: false
+      content_paths:
+        description: If you want to include only specific file set 'true' else 'false'
+        type: string
+        required: false
+
+concurrency:
+  group: ${{ github.workflow }}-cd
+  cancel-in-progress: false
+
+env:
+  BUNDLE_NAME: bundle
+  CONTENT_PATH: ${{ inputs.content_paths == '' && 'false' || 'true' }}
+
+jobs:
+  build:
+    name: Build Artifact
+    runs-on: ubuntu-22.04
+    env:
+      WORKSPACE: ${{ inputs.workspace_name }}
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        name: Checkout
+
+      - name: Prune
+        run: npx turbo@1.13.3 prune --scope ${{ env.WORKSPACE }}
+
+      - name: Setup yarn
+        run: corepack enable
+
+      - name: Setup Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
+        with:
+          node-version-file: ".node-version"
+          cache: "yarn"
+          cache-dependency-path: "./out/yarn.lock"
+
+      - name: Install dependencies
+        run: yarn install --immutable
+        working-directory: ./out
+      
+      # This variable have the form "key=value" e.g.
+      # VAR_1=VALUE_1
+      # VAR_2=VALUE_2
+      # VAR_N=VALUE_N
+      - name: Define env-config.js
```

Comment:
It is not necessary, in fact if the variable `custom_env_config_variables` is not specified the step is not executed; it is useful in case you want to dynamically define the `env-config.js` file